### PR TITLE
Start vivisecting gps_importer plugin 

### PR DESCRIPTION
**Pull request by nyalldawson**
Move the babel format classes to core, cleaning these up and removing unused code and modernizing the API.

### CODE REVIEW

**Comment by nyalldawson** [comment]
I've added unit tests now, because it just felt wrong not to! :laughing:

**Review by nirvn — APPROVED** [review]

File: src/core/gps/qgsbabelformat.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsbabelformat.h"
+
+#include <QRegExp>
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/core/gps/qgsbabelformat.cpp]: Fixed

File: src/plugins/gps_importer/qgsgpsplugin.cpp [hunk 2]
```diff
@@ -422,76 +411,77 @@ void QgsGpsPlugin::setupBabel()
...
 
   // and the GPS devices
   mDevices[QStringLiteral( "Garmin serial" )] =
-    new QgsGpsDevice( QStringLiteral( "%babel -w -i garmin -o gpx %in %out" ),
-                      QStringLiteral( "%babel -w -i gpx -o garmin %in %out" ),
-                      QStringLiteral( "%babel -r -i garmin -o gpx %in %out" ),
-                      QStringLiteral( "%babel -r -i gpx -o garmin %in %out" ),
-                      QStringLiteral( "%babel -t -i garmin -o gpx %in %out" ),
-                      QStringLiteral( "%babel -t -i gpx -o garmin %in %out" ) );
+    new QgsBabelGpsDeviceFormat( QStringLiteral( "%babel -w -i garmin -o gpx %in %out" ),
```
Comment (lbartoletti) [inline src/plugins/gps_importer/qgsgpsplugin.cpp:479]: Maybe replace '-w' with Qgis::GpsFeatureType::Waypoint? same for '-t', '-r'

**Review by lbartoletti — APPROVED** [review]

File: src/plugins/gps_importer/qgsgpsplugin.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/plugins/gps_importer/qgsgpsplugin.cpp:479]: This is a string though - i don't understand how it could be used here

Comment (lbartoletti) [reply src/plugins/gps_importer/qgsgpsplugin.cpp:479]: it's not possible with `QgsAbstractBabelFormat::featureTypeToArgument`?

Comment (nyalldawson) [reply src/plugins/gps_importer/qgsgpsplugin.cpp:479]: I'll do that in a follow up PR -- this code is being moved shortly

Comment (nyalldawson) [reply src/plugins/gps_importer/qgsgpsplugin.cpp:479]: Actually I tried this change, but it ended up much more verbose and less readable as a result

Comment (lbartoletti) [reply src/plugins/gps_importer/qgsgpsplugin.cpp:479]: You're right. It's ok for me.

**Pull request merged**